### PR TITLE
feat: support flattening

### PR DESCRIPTION
**Pull request by latonz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,9 +53,16 @@ To create deep copies, set the `UseDeepCloning` property on the `MapperAttribute
 
 On each mapping method declaration property mappings can be customized.
 If a property on the target has a different name than on the source, the `MapPropertyAttribute` can be applied.
-Flattening is not yet supported.
 If a property should be ignored, the `MapperIgnoreAttribute` can be used.
 
+##### Flattening and unflattening
+
+It is pretty common to flatten objects during mapping. Eg. `Car.Make.Id => Car.MakeId`.
```
Comment (CommonGuy) [inline README.md:63]: Would remove the period before the Eg.: 
> [...] objects during mapping, e.g. `Car.Make.Id => Car.MakeId`.

File: README.md [hunk 2]
```diff
@@ -53,9 +53,16 @@ To create deep copies, set the `UseDeepCloning` property on the `MapperAttribute
 
 On each mapping method declaration property mappings can be customized.
 If a property on the target has a different name than on the source, the `MapPropertyAttribute` can be applied.
-Flattening is not yet supported.
 If a property should be ignored, the `MapperIgnoreAttribute` can be used.
 
+##### Flattening and unflattening
```
Comment (CommonGuy) [inline README.md:61]: The hierarchy of the titles seems odd. Currently, "Properties" and "Flattening and unflattening" are both children of the "Copy behaviour" title... IMO they should all use the same header hierarchy

File: README.md [hunk 3]
```diff
@@ -53,9 +53,16 @@ To create deep copies, set the `UseDeepCloning` property on the `MapperAttribute
...
+
+It is pretty common to flatten objects during mapping. Eg. `Car.Make.Id => Car.MakeId`.
+Mapperly tries to figure out flattenings automatically by making use of the pascal case c# notation.
+If Mapperly can't resolve the target or source property correctly, it is possible to manually configure it by applying the `MapPropertyAttribute`
+by either using the source and target property path names as arrays or using a dot separated property access path string (eg. `Car.Make.Id`).
```
Comment (CommonGuy) [inline README.md:66]: I would provide concrete example here, by replacing "(eg. `Car.Make.Id`)" with
> (e.g. `[MapProperty(Source = new[] { nameof(Car), nameof(Car.Make), nameof(Car.Make.Id) }, Target = new[] { nameof(Car), nameof(Car.MakeId) })]` or `[MapProperty(Source = "Car.Make.Id", Target = "Car.MakeId")]`)

Comment (latonz) [reply README.md]: Adjusted.

File: README.md (hunk 2 above)
Comment (latonz) [reply README.md]: Adjusted.

File: README.md (hunk 1 above)
Comment (latonz) [reply README.md]: Adjusted.

**Review by CommonGuy — APPROVED** [review]

**Pull request merged**